Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -14513,6 +14513,71 @@ void SpeculativeJIT::compileNewTypedArray(Node* node)
     }
 }
 
+void SpeculativeJIT::compileNewTypedArrayBuffer(Node* node)
+{
+    switch (node->child1().useKind()) {
+    case Int32Use: {
+#if USE(LARGE_TYPED_ARRAYS)
+        // The operations we call on the slow path expect a intptr_t, so int64_t on 64 bit platforms
+        SpeculateInt32Operand size(this, node->child1());
```

Comment:
Is that possible `node->child1()` can be a constant. If that's the case, then we can have a fast path here right?

Reply:
If I am understanding this correctly, `ByteCodeParser::handleConstantFunction` should emit the node for 
```
new ArrayBuffer(10)
```
only but not
```
new ArrayBuffer(x)
```
. So, `node->child1()` must be a constant?

Reply:
Ah, no. It is generally just handling any inputs. And we do type speculation. So it can be constant, or not constant.